modules/core/src/main/java/org/apache/ignite/internal/util/Cursor.java

```diff
@@ -25,5 +25,38 @@
  * @param <T> Type of elements.
  */
 public interface Cursor<T> extends Iterator<T>, Iterable<T>, AutoCloseable {
+    /** {@inheritDoc} */
+    @Override
+    default Iterator<T> iterator() {
+        return this;
+    }
 
+    /**
+     * Creates an iterator based cursor.
+     *
+     * @param it Iterator.
+     * @param <T> Type of elements in iterator.
+     * @return Cursor.
+     */
+    static <T> Cursor<T> fromIterator(Iterator<? extends T> it) {
+        return new Cursor<T>() {
+            /** {@inheritDoc} */
+            @Override
+            public void close() throws Exception {
+                // No-op.
```

Comment:
Perhaps, it would be better to propagate `close()` to the iterator in case the latter implements an interface `AutoCloseable`

Reply:
Nice, I'll do it